src/components/ProtectedRoute.tsx

```diff
@@ -0,0 +1,14 @@
+import React, { FC } from "react";
+import { Navigate, Route, RouteProps } from "react-router-dom";
+
+interface Props extends RouteProps {
+  isAuth: boolean;
+}
+const ProtectedRoute: FC<Props> = ({ isAuth, ...routerProps }: Props) => {
```

Comment:
Although your `ProtectedRoute` function is perfectly fine. I would encourage you to make more use of the syntax Javascript/Typescript provides. It'll help you shorten your code as well! For example, you could rewrite you `ProtectedRoute` function in the following manner: 

`const ProtectedRoute: FC<Props> = ({ isAuth, ...routerProps }: Props) =>  isAuth ? (<Route {...routerProps} />) : (<Navigate to="/" />);`